pkg/apis/hive/v1alpha1/hiveconfig_types.go

```diff
@@ -32,6 +44,32 @@ type HiveConfigStatus struct {
 	AggregatorClientCAHash string `json:"aggregatorClientCAHash,omitempty"`
 }
 
+// ExternalDNSConfig contains settings for running external-dns in a Hive
+// environment.
+type ExternalDNSConfig struct {
+
+	// Image is a reference to the image that will run the external-dns controller.
+	// If not specified, a default image will be used.
+	Image string `json:"image,omitempty"`
```

Comment:
Just a comment, we can't use this for hive-stage as we don't have perms to edit it, and we'd need it to potentially be automatically rolled out via jenkins, so we'd probably use the deployment.yaml itself right now? 

Image override might be useful for someone external using hive, but then I'm not sure, I'm not totally clear on how all these images need to get pinned and versioned just yet.

Reply:
If we can't edit hiveconfig how do we configure managed dns?